tests/containers/kubectl.pm

```diff
@@ -0,0 +1,139 @@
+    assert_script_run('curl -o deployment.yml ' . data_url('containers/kubectl/deployment.yml'));
+    assert_script_run('kubectl apply -f deployment.yml');
+    assert_script_run('kubectl wait deployment nginx-deployment --for condition=available --timeout=300s', timeout => 330);
+    validate_script_output('kubectl describe deployments/nginx-deployment | grep Replicas', sub { $_ =~ m/2 desired.*2 total/ });
+    assert_script_run('kubectl describe deployments/nginx-deployment');
+    # Scale out
+    assert_script_run('kubectl scale --replicas=5 deployments/nginx-deployment');
+    assert_script_run('kubectl wait deployment nginx-deployment --for condition=available --timeout=300s', timeout => 330);
+    validate_script_output('kubectl describe deployments/nginx-deployment | grep Replicas', sub { $_ =~ m/5 desired.*5 total/ });
+    # Scale in
+    assert_script_run('kubectl scale --replicas=2 deployments/nginx-deployment');
+    assert_script_run('kubectl wait deployment nginx-deployment --for condition=available --timeout=300s', timeout => 330);
+    validate_script_output('kubectl describe deployments/nginx-deployment | grep Replicas', sub { $_ =~ m/2 desired.*2 total/ });
+    # Test the port-forwarding and the webserver
+    my $pid = background_script_run('kubectl port-forward deploy/nginx-deployment 8008:80');
+    sleep(10);    # I don't see a better way to wait for the forward to be ready :-(
+    validate_script_output("curl http://localhost:8008/index.html", qr/I am groot/);
+    assert_script_run("kill $pid");    # terminate port-forwarding
+
+    ## Test service
+    # Create predefined service, and check if it appears as desired
+    record_info('Test 5', 'service test');
+    assert_script_run('curl -o service.yml ' . data_url('containers/kubectl/service.yml'));
+    assert_script_run('kubectl apply -f service.yml');
+    validate_script_output('kubectl get services', qr/web-load-balancer/);
+    validate_script_output('kubectl describe services/web-load-balancer', sub { $_ =~ m/.*Name:.*web-load-balancer.*/ });
+    validate_script_output('kubectl describe services/web-load-balancer', sub { $_ =~ m/.*Port:.*8080\/TCP.*/ });
+    validate_script_output('kubectl describe services/web-load-balancer', sub { $_ =~ m/.*TargetPort:.*80\/TCP.*/ });
+    validate_script_output('kubectl describe services/web-load-balancer', sub { $_ =~ m/.*Endpoints:.*10.*/ });
+    validate_script_output("curl http://localhost:8080/index.html", qr/I am groot/);
+    assert_script_run('kubectl delete -f service.yml');
+
+    assert_script_run('kubectl delete -f deployment.yml');
+}
+
+sub post_fail_hook {
+    my ($self) = @_;
+
+    # Try to collect as much information about kubernetes as possible
+    script_run('kubectl describe deployments');
+    script_run('kubectl describe services');
+    script_run('kubectl describe pods');
+    # Cleanup
+    script_run('kubectl delete -f service.yml');
+    script_run('kubectl delete -f deployment.yml');
+}
```

Comment:
I guess we are not deleting k3s because is not necessary. The VM will be destroyed.